3-networks/envs/shared/net-hubs.tf

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  base_net_hub_project_id           = try(data.google_projects.base_net_hub["yes"].projects[0].project_id, null)
+  restricted_net_hub_project_id     = try(data.google_projects.restricted_net_hub["yes"].projects[0].project_id, null)
+  restricted_net_hub_project_number = try(data.google_projects.restricted_net_hub["yes"].projects[0].number, null)
+}
+
+/******************************************
+  Base Network Hub Project
+*****************************************/
+
+data "google_projects" "base_net_hub" {
+  for_each = var.hub_and_spoke ? toset(["yes"]) : toset([])
+  filter   = "parent.id:${split("/", data.google_active_folder.common.name)[1]} labels.application_name=org-base-net-hub lifecycleState=ACTIVE"
+}
+
+/******************************************
+  Restricted Network Hub Project
+*****************************************/
+
+data "google_projects" "restricted_net_hub" {
+  for_each = var.hub_and_spoke ? toset(["yes"]) : toset([])
+  filter   = "parent.id:${split("/", data.google_active_folder.common.name)[1]} labels.application_name=org-restricted-net-hub lifecycleState=ACTIVE"
+}
+
+/******************************************
+  Base Network VPC
+*****************************************/
+
+module "base_shared_vpc" {
+  source                        = "../../modules/base_shared_vpc"
+  for_each                      = var.hub_and_spoke ? toset(["yes"]) : toset([])
+  project_id                    = local.base_net_hub_project_id
+  environment_code              = "c"
+  org_id                        = var.org_id
+  parent_folder                 = var.parent_folder
+  bgp_asn_subnet                = "64514"
+  default_region1               = var.default_region1
+  default_region2               = var.default_region2
+  domain                        = var.domain
+  windows_activation_enabled    = var.base_hub_windows_activation_enabled
```

Comment:
Would we need these if we are not deploying workloads in hub?

Reply:
We'll deploy the appliances for transitive peering in the hub.

Reply:
(but they are not windows) :)

Reply:
@drebes sg, I am not super familiar with the requirements for appliances, feel free to call out anything in the PR that we should add/remove.

Reply:
I would keep that as an option, it's quite often we have some "common" workloads in hub, like spinnaker/jenkins/vault or even Win domain controller.